dali/test/python/test_pipeline.py

```diff
@@ -600,6 +600,67 @@ def define_graph(self):
             np_transposed = images[b].transpose((2, 0, 1))
             np_transposed = np.ascontiguousarray(np_transposed)
             assert(np.array_equal(np_transposed, images_transposed[b]))
+def test_equal_nvJPEGDecoderCrop_nvJPEGDecoder():
+    batch_size =128
+
+    class HybridPipe(Pipeline):
+        def __init__(self, batch_size, num_threads, device_id, num_gpus):
+            super(HybridPipe, self).__init__(batch_size,
+                                             num_threads,
+                                             device_id)
+            self.input = ops.CaffeReader(path = caffe_db_folder, shard_id = device_id, num_shards = num_gpus)
+            self.decode = ops.nvJPEGDecoder(device = "mixed", output_type = types.RGB)
+            self.pos_rng_x = ops.Uniform(range = (0.0, 1.0), seed=1234)
+            self.pos_rng_y = ops.Uniform(range = (0.0, 1.0), seed=5678)
+            self.crop = ops.Crop(device="gpu", crop =(224,224))
+
+        def define_graph(self):
+            self.jpegs, self.labels = self.input()
+
+            pos_x = self.pos_rng_x()
+            pos_y = self.pos_rng_y()
+            images = self.decode(self.jpegs)
+            crop = self.crop(images, crop_pos_x=pos_x, crop_pos_y=pos_y)
+            return (crop, self.labels)
+
+
+        def iter_setup(self):
+            pass
+
+    pipe = HybridPipe(batch_size=batch_size, num_threads=1, device_id = 0, num_gpus = 1)
+    class HybridPipe1(Pipeline):
+        def __init__(self, batch_size, num_threads, device_id, num_gpus):
+            super(HybridPipe1, self).__init__(batch_size,
+                                             num_threads,
+                                             device_id)
+            self.input = ops.CaffeReader(path = caffe_db_folder, shard_id = device_id, num_shards = num_gpus)
+            self.pos_rng_x = ops.Uniform(range = (0.0, 1.0), seed=1234)
+            self.pos_rng_y = ops.Uniform(range = (0.0, 1.0), seed=5678)
+            self.decode = ops.nvJPEGDecoderCrop(device = 'mixed', output_type = types.RGB, crop = (224, 224))
+
+        def define_graph(self):
+            self.jpegs, self.labels = self.input()
+            pos_x = self.pos_rng_x()
+            pos_y = self.pos_rng_y()
+            images = self.decode(self.jpegs, crop_pos_x=pos_x, crop_pos_y=pos_y)
+            return (images, self.labels)
+
+        def iter_setup(self):
+            pass
+    pipe.build()
+    pipe_out = pipe.run()
+    pipe1 = HybridPipe1(batch_size=batch_size,
+                          num_threads=1,
+                          device_id = 0, num_gpus = 1)
+    pipe1.build()
+    pipe_out1 = pipe1.run()
+    for i in range(batch_size):
+        pipe_out_cpu = pipe_out[0].as_cpu()
+        pipe_out_cpu1 = pipe_out1[0].as_cpu()
+        assert(np.sum(np.abs(pipe_out_cpu.at(i)-pipe_out_cpu1.at(i)))==0)
```

Comment:
```suggestion
        assert(np.sum(pipe_out_cpu.at(i)-pipe_out_cpu1.at(i))==0)
```
In case you'd like to compare the sum to 0, `np.abs` is redundant here.

Reply:
it is not redundant. If you have a diff `-1` and and other `1` they'd cancel out and that's not what you want. Please bring back `np.abs`